xonsh/base_shell.py

```diff
@@ -339,8 +339,12 @@ def default(self, line):
                 hist.last_cmd_rtn = 1  # return code for failure
         finally:
             ts1 = ts1 or time.time()
-            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee.getvalue())
+            tee_out = tee.getvalue()
+            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee_out)
             self.accumulated_inputs += src
+            append_newline = env.get('XONSH_APPEND_NEWLINE')
+            if append_newline and not tee_out.endswith(os.linesep):
+                print(append_newline, end='')
```

Comment:
> isn't this suppose to print a new line and not the envvar?
We need to print the envvar, of course.
> you can always use the on_post_cmd event
Can I get previous output and check for os.linesep at the end in this case?